sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/console.scala

```diff
@@ -49,7 +49,7 @@ class ConsoleSink(options: Map[String, String]) extends Sink with Logging {
     println("-------------------------------------------")
     // scalastyle:off println
     data.sparkSession.createDataFrame(
-      data.sparkSession.sparkContext.parallelize(data.collect()), data.schema)
```

Comment:
I think we also need to consume all data to change the internal states in stateful operators. How about this:
```Scala
    val encoder = data.exprEnc.resolveAndBind(
      data.logicalPlan.output,
      data.sparkSession.sessionState.analyzer)

    val numRowsToFetch = numRowsToShow + 1
    val takeResult = data.queryExecution.toRdd.mapPartitions { iter =>
      var numFetched = 0
      val v = ArrayBuffer[Row]()
      while (numFetched < numRowsToFetch && iter.hasNext) {
        v += encoder.fromRow(iter.next())
        numFetched += 1
      }
      // Consume all data to update internal states in stateful operators.
      while (iter.hasNext) {
        iter.next()
      }
      v.iterator
    }.collect().toSeq.take(numRowsToFetch)

    data.sparkSession.createDataFrame(
      data.sparkSession.sparkContext.parallelize(takeResult),
      data.schema).show(numRowsToShow, isTruncated)
```